File: fixtures/scale/dart/medium/lib/ast/unary_expr.dart
import 'expr.dart';

class UnaryExpr implements Expr {
  final String operator;
  final Expr operand;

  UnaryExpr(this.operator, this.operand);

  @override
  String describe() => '$operator${operand.describe()}';

  bool get isNegation => operator == '-';
}

File: fixtures/scale/java/medium/src/calc/ast/UnaryExpr.java
package calc.ast;

public class UnaryExpr implements Expr {
    private final char operator;
    private final Expr operand;

    public UnaryExpr(char operator, Expr operand) {
        this.operator = operator;
        this.operand = operand;
    }

    public char getOperator() {
        return operator;
    }

    public Expr getOperand() {
        return operand;
    }
}
